validator/sawtooth_validator/networking/handlers.py

```diff
@@ -88,3 +132,74 @@ def handle(self, connection_id, message_content):
             HandlerStatus.RETURN,
             message_out=ack,
             message_type=validator_pb2.Message.NETWORK_ACK)
+
+
+class AuthorizationTrustRequestHandler(Handler):
+
+    def __init__(self, network, permission_verifier):
+        self._network = network
+        self._permission_verifier = permission_verifier
+
+    def handle(self, connection_id, message_content):
+        if self._network.get_connection_status(connection_id) != \
+                ConnectionStatus.CONNECTION_REQUEST:
+            LOGGER.debug("Connection has previous message was not a"
+                         " ConnectionRequest, Remove connection to %s",
+                         connection_id)
+            violation = AuthorizationViolation(
+                violation=AuthorizationViolation.NETWORK)
+            return HandlerResult(
+                HandlerStatus.RETURN_AND_CLOSE,
+                message_out=violation,
+                message_type=validator_pb2.Message
+                .AUTHORIZATION_VIOLATION)
+
+        request = AuthorizationTrustRequest()
+        request.ParseFromString(message_content)
+
+        # Check that the connection's public key is allowed by the network role
+        roles = self._network.roles
+        for role in request.requested_roles:
+            if role == request.NETWORK or role == request.ALL:
+                permitted = False
+                if "network" in roles:
+                    permitted = self._permission_verifier.check_network_role(
+                        request.public_key)
+                if not permitted:
+                    violation = AuthorizationViolation(
+                        violation=AuthorizationViolation.NETWORK)
+                    return HandlerResult(
+                        HandlerStatus.RETURN_AND_CLOSE,
+                        message_out=violation,
+                        message_type=validator_pb2.Message
```

Comment:
If splitting on the dot is necessary for line width, can you put in a \ to alert the reader that there is a line break.

Reply:
That would cause the lint error: the backslash is redundant between brackets